crates/client/rpc/src/trace_api.rs

```diff
@@ -240,6 +246,8 @@ fn tx_execution_infos_to_simulated_transactions<B: BlockT>(
                     })
                     .transpose()?;
 
+                let state_diff = blockifier_to_rpc_state_diff_types(state_diff.clone()).ok();
```

Comment:
You both clone and shadow the var `state_diff`. Is there a way you could just take ownership of it?

Reply:
Maybe using `into_iter` there `tx_types.iter().zip(transaction_execution_results.iter())`